lib/tramway/forms/normalizations.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Tramway
+  module Forms
+    # This is the same `normalizes` feature like in Rails
+    # https://api.rubyonrails.org/v7.1/classes/ActiveRecord/Normalization/ClassMethods.html#method-i-normalizes
+    module Normalizations
+      # A collection of methods that would be using in users forms
+      module ClassMethods
+        # :reek:BooleanParameter { enabled: false }
+        def normalizes(*attributes, with:, apply_to_nil: false)
+          attributes.each do |attribute|
+            @normalizations.merge!(attribute => { proc: with, apply_to_nil: })
+          end
+        end
+
+        def normalizations
+          __ancestor_normalizations.merge(@normalizations)
+        end
+
+        # :reek:ManualDispatch { enabled: false }
+        def __ancestor_normalizations(klass = superclass)
+          superklass = klass.superclass
+
+          return {} unless superklass.respond_to?(:normalizations)
+
+          klass.normalizations.merge!(__ancestor_normalizations(superklass))
+        end
+
+        # :reek:UtilityFunction { enabled: false }
+        def __initialize_normalizations(subclass)
+          subclass.instance_variable_set(:@normalizations, {})
+        end
+      end
+
+      def self.included(base)
+        base.extend ClassMethods
+      end
+
+      def __apply_normalizations(params)
+        self.class.normalizations.reduce(params) do |hash, (attribute, normalization)|
+          if hash.key?(attribute) || normalization[:apply_to_nil]
```

Comment:
@kalashnikovisme it seems `apply_to_nil` is working here not like in rails (and it seems there is no tests for it?)

if you have `{ a: nil }`, `hash.key?(:a)` will return `true`, and normalization will be called.
But it should be called in this case only if `apply_to_nil` is true

Reply:
Nice catch!!!!

Explored tests, and yeah they check this case wrong. Will fix it

Reply:
PR with a [fix](https://github.com/Purple-Magic/tramway/pull/50)